_pytest/cacheprovider.py

```diff
@@ -14,7 +14,8 @@
 class Cache(object):
     def __init__(self, config):
         self.config = config
-        self._cachedir = config.rootdir.join(".cache")
+        cache_dirname = config.getini("cache_dirname")
+        self._cachedir = config.rootdir.join(cache_dirname)
```

Comment:
@RockBomber good point, in that case, how about warn/fail in case its an absolute path or it starts with ~ until someone implements a more complete solution